### PR TITLE
feat(sdk): add new endpoints

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes https://linear.app/nango/issue/NAN-1753/document-new-endpoints-integrations-and-providers

- Add new endpoints to the SDK
There is only one method that was different enough, queries and output that even with signature overload it wasn't possible to make it non-breaking. Honestly, I didn't find anything compelling to deprecate without either being ugly or breaking customers' scripts, so very open to suggestions. 

- Document new SDK method
- Add support for `include=webhook` in `GET /integrations/:key`
The SDK needs it, and some customers use it. It's a secret but not credentials, so I didn't put a too vague name. If we need credentials at some point, we can differentiate.

### CODE REVIEW

File: docs-v2/reference/sdks/node.mdx [hunk 1]
```diff
@@ -94,6 +157,42 @@ await nango.listIntegrations()
 
 Returns a specific integration.
 
+```js
+await nango.getIntegrationV2(<UNIQUE_KEY>, { include?: ['webhook'] });
```
Comment (TBonnin) [inline docs-v2/reference/sdks/node.mdx]: what about having the new method take an single options object instead, then signature is different and we can use method overloading.
```
type getIntegrationOptions = {uniqueKey: string, query?: ...}

public async getIntegration({uniqueKey, query} : getIntegrationOptions);
public async getIntegration(name: string);
public async getIntegration(nameOrOptions: string | getIntegrationOptions) {
...
```
Same for `getProvider({name}: { name: string})` to keep it consistent

File: packages/node-client/lib/utils.ts [hunk 2]
```diff
@@ -40,3 +40,19 @@ export function getUserAgent(userAgent?: string): string {
...
+        return;
+    }
+
+    Object.entries(queries).forEach(([name, value]) => {
+        if (Array.isArray(value)) {
+            for (const el of value) {
+                url.searchParams.set(name, el);
+            }
+        } else {
+            url.searchParams.set(name, value || '');
```
Comment (TBonnin) [inline packages/node-client/lib/utils.ts]: should the query param be set if value is undefined/null?

File: packages/server/lib/controllers/integrations/uniqueKey/getIntegration.ts [hunk 3]
```diff
@@ -12,10 +12,25 @@ export const validationParams = z
...
+    .object({
+        include: z
+            .union([
+                z
+                    .string()
+                    .transform((v) => [v])
+                    .pipe(z.array(valInclude)),
+                z.array(valInclude)
+            ])
+            .optional()
```
Comment (TBonnin) [inline packages/server/lib/controllers/integrations/uniqueKey/getIntegration.ts:21]: is something like this easier to read?
```
const validationQuery = z.object({
  include: z
    .union([valInclude, z.array(valInclude)])
    .optional()
    .transform((val) => (Array.isArray(val) ? val : val ? [val] : []))
});
```

File: docs-v2/reference/sdks/node.mdx (hunk 1 above)
Comment (bodinsamuel) [reply docs-v2/reference/sdks/node.mdx]: Could work indeed, it's a paradigm shift though. Usually in Nodejs SDK the first param to get() is an id, string or number not an object.

We can have naming consistency:
```ts
- getProvider({ id })
- getIntegration({ id })
- getConnection({ id })
```
or still use the name
```ts
- getProvider({ name })
- getIntegration({ unique_key })
- getConnection({ connection_id })
```

wdyt?

File: packages/node-client/lib/utils.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/node-client/lib/utils.ts]: yeah good point (I'll still allow false which is a legitimate value)

File: packages/server/lib/controllers/integrations/uniqueKey/getIntegration.ts (hunk 3 above)
Comment (bodinsamuel) [reply packages/server/lib/controllers/integrations/uniqueKey/getIntegration.ts:21]: 👌🏻

File: packages/node-client/bin/get-integration.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ const args = process.argv.slice(2);
 const nango = new Nango({ host: 'http://localhost:3003', secretKey: args[0] });
 
 nango
-    .getIntegration(args[1], args[2])
+    .getIntegrationV2(args[1])
```
Comment (khaliqgant) [inline packages/node-client/bin/get-integration.js]: Would rather this be a separate file until we actually retire the original method

**Review by khaliqgant — APPROVED** [review]
non blocking comment inline

File: docs-v2/reference/sdks/node.mdx (hunk 1 above)
Comment (TBonnin) [reply docs-v2/reference/sdks/node.mdx]: I prefer option 2. It is a bit more self-explanatory what the param is.

**Comment by bodinsamuel** [comment]
@TBonnin Updated with signature change, thanks for the suggestion 💪🏻

**Review by TBonnin — APPROVED** [review]

**Pull request merged**